webui/src/lib/onboarding/repoOnboardingService.ts

```diff
@@ -0,0 +1,119 @@
+import { OnBoardingSteps } from "./types";
+import { NavigateFunction } from "react-router-dom";
+import { branches, retention, branchProtectionRules, objects } from "../api";
+
+const IMPORT_BRANCH_REGEX = /^_(.*)_imported$/;
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_FILE_NAME = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch = "main"
+): Promise<boolean> => {
+  const objectList = await objects.list(repo, defaultBranch, "");
+  return objectList.results.some(
+    // when the API client is typed, we can remove the eslint-disable
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (object: any) => object?.path === SAMPLE_REPO_FILE_NAME
+  );
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch = "main"
+): Promise<boolean> => {
+  const promises = [];
+  promises.push(retention.setGCPolicyPreflight(repoId));
+  promises.push(branchProtectionRules.createRulePreflight(repoId));
+  promises.push(
+    objects.list(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    )
+  );
+  const results = await Promise.allSettled(promises);
+  if (results.some((result) => result.status === "rejected" || !result.value)) {
+    return false;
+  }
+  return true;
+};
+
+export const getRepoOnboardingSteps = (
+  currentRepo: string,
+  defaultBranch: string,
+  objectStoreName: string,
+  navigate: NavigateFunction
+): OnBoardingSteps => [
+  {
+    id: "import-data",
+    title: "Import your existing data",
+    description: `Import your existing data from ${objectStoreName} to your repository by creating pointers to it (no copying of data involved)`,
+    cta: "Run import",
+    onClick: () =>
+      navigate(`/repositories/${currentRepo}/objects?importDialog=true`),
+    showStep: () => objectStoreName !== "local",
+    isCompleted: async () => {
+      const repoBranches = await branches.list(currentRepo);
+      // when the API client is typed, we can remove the eslint-disable
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      return repoBranches.results.some((branch: any) =>
+        IMPORT_BRANCH_REGEX.test(branch.id)
+      );
+    },
+  },
+  {
+    id: "garbage-collection",
+    title: "Setup garbage collection",
+    description:
+      "Setup rules to automatically manage when data can be deleted from your repository",
```

Comment:
This text **definitely** needs to link to the relevant reference.  After all, nothing happens if you just add GC rules to a repository, you need a Spark job.

Reply:
@ozkatz I think @arielshaqed makes a good point. The text is from the PRD. Want to change the text, or should I take it?